invenio_rdm_records/oaiserver/services/services.py

```diff
@@ -58,6 +61,26 @@ def _get_one(self, raise_error=True, **kwargs):
             errors.append(str(e))
         return set, errors
 
+    def _validate_spec(self, spec):
+        """Checks the validity of the provided spec."""
+        # Reserved for community integration
+        if spec.startswith("community-"):
+            raise ValidationError(
+                "The spec must not start with 'communities-'",
+                field_name="spec",
+            )
+
+        # See https://www.openarchives.org/OAI/openarchivesprotocol.html#Set
+        blop = re.compile(r"[-_.!~*'()\w]+")
+        if not bool(blop .match(spec)):
+            raise ValidationError(
```

Comment:
Question: Is this aligned with ``invenio_communities.communities.services.components:PIDComponent._validate()``. Basically are we gonna end up with some communities having invalid set specs?

Reply:
Regex in communities `PIDComponent` is a subset of this and only allows letters, numbers and dashes.